easybuild/easyblocks/o/opencv.py

```diff
@@ -165,7 +165,11 @@ def configure_step(self):
         # see https://github.com/opencv/opencv/wiki/CPU-optimizations-build-options
         if self.toolchain.options.get('optarch') and 'CPU_BASELINE' not in self.cfg['configopts']:
             optarch = build_option('optarch')
-            if not optarch:
+            if (
+                not optarch
+                or (get_software_root('GCC') and (optarch == 'march=native' or optarch.get('GCC') == 'march=native'))
+                or (get_software_root('intel-compilers') and (optarch == 'xHost' or optarch.get('Intel') == 'xHost'))
```

Comment:
Ideally we don't hardcode `march=native` and `xHost` here, since those values may change.

We should check whether we can extract the default from framework (and if not maybe we need to introduce a constant in framework so the default can be easily obtained without hardcoding)

Reply:
And note that a manually set optarch may well be something more than march=native/xHost so a simple == doesn't suffice

Reply:
@boegel  @akesandgren done in https://github.com/easybuilders/easybuild-easyblocks/pull/2954/commits/7310de7559dd44572a8d7ca338e026387b460b5e